### PR TITLE
Set assets manifest metadata for assets that get shipped with .NET release

### PR DESCRIPTION
**Pull request by MilenaHristova**
Tracking issue: https://github.com/dotnet/release/issues/822

Setting metadata for assets that will get shipped as part of .NET release in the assets manifest file. 
I added the `ProducesDotNetReleaseShippingAssets` property in https://github.com/dotnet/runtime/pull/98665
Since runtime doesn't use arcade's default publishing, this is editing `prepare-artifacts.proj` where the metadata is getting added to the manifest

### CODE REVIEW

File: src/installer/prepare-artifacts.proj [hunk 1]
```diff
@@ -102,36 +104,47 @@
...
       <ItemsToPush Include="$(ArtifactsShippingPackagesDir)runtime-productVersion.txt">
         <RelativeBlobPath>$(InstallersRelativePath)runtime-productVersion.txt</RelativeBlobPath>
         <PublishFlatContainer>true</PublishFlatContainer>
+        <ManifestArtifactData Condition="'$(ProducesDotNetReleaseShippingAssets)' == 'true'">DotNetReleaseShipping=true</ManifestArtifactData>
       </ItemsToPush>
 
       <ItemsToPush Include="@(WorkloadsVSInsertionFile)">
         <RelativeBlobPath>$(InstallersRelativePath)workloads/%(Filename)%(Extension)</RelativeBlobPath>
         <PublishFlatContainer>true</PublishFlatContainer>
+        <ManifestArtifactData Condition="'$(ProducesDotNetReleaseShippingAssets)' == 'true'">DotNetReleaseShipping=true</ManifestArtifactData>
```
Comment (ViktorHofer) [inline src/installer/prepare-artifacts.proj:147]: Consider using an ItemDefinitionGroup instead of all these manual metadata updates.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
I assume the same change needs to be done in windowsdesktop and deployment-tools?

**Comment by MilenaHristova** [comment]
> I assume the same change needs to be done in windowsdesktop and deployment-tools?

yes, for windowsdesktop: https://github.com/dotnet/windowsdesktop/pull/4173
deployment-tools is not on the gather-drop list, I'll have to check about it

**Pull request merged**